packages/avs-bot/src/main.ts

```diff
@@ -0,0 +1,335 @@
+  timestamp: number;
+  userId: string;
+}
+
+class App {
+  storagePath = path.join(process.cwd(), '.tmp', WIRE_EMAIL || '');
+  storeOptions = {fileExtension: '.json'};
+  storeEngine: CRUDEngine = new FileEngine(this.storagePath, this.storeOptions);
+  wUser?: number;
+  wCall?: Wcall;
+  cookie?: Cookie;
+  activeCalls: {[key: string]: Call} = {};
+  account?: Account;
+  mediaStream: MediaStream = new MediaStream();
+  audioSource: RTCAudioSourceWrapper = new RTCAudioSourceWrapper();
+
+  public async start() {
+    await this.storeEngine.init(this.storagePath, this.storeOptions);
+    const config = {
+      store: this.storeEngine,
+      urls: APIClient.BACKEND.STAGING,
+    };
+    const apiClient = new APIClient(config);
+    const account = new Account(apiClient, async (): Promise<CRUDEngine> => this.storeEngine);
+    this.account = account;
+
+    apiClient.on(APIClient.TOPIC.ACCESS_TOKEN_REFRESH, async (accessToken: AccessTokenData) => {
+      await this.storeEngine.updateOrCreate(AUTH_TABLE_NAME, AUTH_ACCESS_TOKEN_KEY, accessToken);
+    });
+
+    apiClient.on(APIClient.TOPIC.COOKIE_REFRESH, async (cookie?: Cookie) => {
+      console.log('new cookie', cookie);
+      if (!cookie) {
+        return;
+      }
+      this.cookie = cookie;
+      const entity = {expiration: cookie.expiration, zuid: cookie.zuid};
+      await this.storeEngine.updateOrCreate(AUTH_TABLE_NAME, AUTH_COOKIE_KEY, entity);
+    });
+
+    await this.getCookie();
+
+    let userId: string;
+    try {
+      if (!this.cookie) {
+        throw new Error('No cookie found');
+      }
+      userId = (await account.init(CLIENT_TYPE, this.cookie)).userId;
+    } catch (error) {
+      console.log('Failed to get new access token with cookie');
+      userId = (await account.login(loginData)).userId;
+    }
+    const {
+      localClient: {id: clientId},
+    } = await account.initClient(loginData);
+    await this.initAvs(userId, clientId);
+
+    account.on(PayloadBundleType.CALL, call => {
+      this.onIncomingCallMessage(call);
+    });
+
+    await account.listen();
+  }
+
+  onIncomingCallMessage = (callMessage: CallMessage) => {
+    try {
+      const callContent: CallingContent = callMessage.content as CallingContent;
+      if (!callMessage.fromClientId) {
+        throw new Error('callMessage.fromClientId not found');
+      }
+      const avsResult = this.wCall!.recvMsg(
+        this.wUser!,
+        callContent,
+        callContent.length,
+        new Date().getSeconds(),
+        new Date(callMessage.timestamp).getSeconds(),
+        callMessage.conversation,
+        callMessage.from,
+        callMessage.fromClientId,
+      );
+      if (avsResult !== 0) {
+        throw new Error(`recvMsg failed with code "${avsResult}"`);
+      }
+      this.answerCall(callMessage);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  answerCall = (call: CallMessage) => {
+    if (!this.wCall || !this.wUser) {
+      return;
```

Comment:
Throw an error here because it is an undefined state.

Reply:
https://github.com/wireapp/wire-web-packages/pull/3070/commits/7cb56da3433526551633eefd9d2ec8504249fccf